### PR TITLE
fix playlist paging for externally loaded playlists with possible custom params (TR-408)

### PR DESCRIPTION
**Pull request by vladm3**

### CODE REVIEW

File: modules/KalturaSupport/components/playlistAPI.js [hunk 1]
```diff
@@ -190,6 +192,14 @@
 						'pager:pageSize': _this.getConfig('pageSize'),
 						'id': _this.playlistSet[_this.currentPlaylistIndex].id
 					};
+
+					// extend with paging props
+					$.extend(playlistRequest, {
```
Comment (amirch1) [inline modules/KalturaSupport/components/playlistAPI.js]: @vladm3 - this code is redundant if not using saved params (when creating a new request object).

File: modules/KalturaSupport/components/playlistAPI.js [hunk 2]
```diff
@@ -182,7 +182,9 @@
 					_this.pagingInProgress = true;
 					_this.page++; // move to next page
 					mw.log("Playlist:: scrollEnd event. paging: true: trying to load next playlist page: page="+_this.page+", pageSize"+_this.getConfig('pageSize'));
-					var playlistRequest = {
+
```
Comment (amirch1) [inline modules/KalturaSupport/components/playlistAPI.js]: @vladm3 - for readability and prevent redundant $.extend call, please separate the 2 requests creation:

if ( _this.playlistSet[_this.currentPlaylistIndex].playlistParams){
// use saved params and use $.extend
}else{
// create the original request
}

File: modules/KalturaSupport/components/playlistAPI.js (hunk 1 above)
Comment (vladm3) [inline modules/KalturaSupport/components/playlistAPI.js]: @amirch1 right, it is actually redundant when not using saved params

File: modules/KalturaSupport/components/playlistAPI.js (hunk 2 above)
Comment (vladm3) [inline modules/KalturaSupport/components/playlistAPI.js]: @amirch1 It's possible to resolve redundant call if remove L190-L192, but, I'll update file to use branching, like you've mentioned

**Comment by vladm3** [comment]
@amirch1 - updated PR

**Pull request merged**